flink-streaming-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaConsumerTestBase.java

```diff
@@ -482,12 +476,39 @@ public void runStartFromEarliestOffsets() throws Exception {
 	 * ignores the "auto.offset.reset" behaviour as well as any committed group offsets in Kafka.
 	 */
 	public void runStartFromLatestOffsets() throws Exception {
```

Comment:
To make this test easier without having to sleep, the test now does this:

1. First write 50 records to each partition (these shouldn't be read)
2. Set some offsets in Kafka (should be ignored)
3. Start a latest-reading consuming job. This jobs throws exception if it reads any of the first 50 records
4. Wait until the consume job has fully started (added an util method to `JobManagerCommunicationUtils` for this)
5. Write 200 extra records to each partition.
6. Once the writing finishes, cancel the consume job.
7. Check if the consume job threw any test errors.